pkg/controller/replicaschedulingpreference/controller.go

```diff
@@ -15,6 +14,727 @@ See the License for the specific language governing permissions and
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/federation/v1alpha1"
+	fedclientset "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset_generated/clientset"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util/planner"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util/podanalyzer"
+	apiv1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/client-go/util/workqueue"
+	crclientset "k8s.io/cluster-registry/pkg/client/clientset_generated/clientset"
+)
+
+const (
+	allClustersKey = "ALL_CLUSTERS"
+)
+
+// ReplicaSchedulingPreferenceController syncronises the template, override
+// and placement for a target deployment template with its spec (user preference).
+// TODO: make this usable atleast for current known scheduling types (deployments and replicasets)
+type ReplicaSchedulingPreferenceController struct {
+	// Used to allow time delay in triggering reconciliation
+	// when any of RSP, target template, override or placement
+	// changes.
+	deliverer *util.DelayingDeliverer
+
+	// For triggering reconciliation of all resources (only in
+	// federation). This is used when a new cluster becomes available.
+	clusterDeliverer *util.DelayingDeliverer
+
+	// Contains target resources present in members of federation.
+	targetInformer util.FederatedInformer
+
+	// Informs about pods present in members of federation.
+	// TODO: Rather then a separate informer, a typed client
+	// interface can be added into the targetInformer.
+	podInformer util.FederatedInformer
+
+	// Store for self (ReplicaSchedulingPreference)
+	store cache.Store
+	// Informer for self (ReplicaSchedulingPreference)
+	controller cache.Controller
+	// Client to the federation API
+	fedClient fedclientset.Interface
+
+	// Store for the templates of the federated type
+	templateStore cache.Store
+	// Informer for the templates of the federated type
+	templateController cache.Controller
+
+	// Store for the override directives of the federated type
+	overrideStore cache.Store
+	// Informer controller for override directives of the federated type
+	overrideController cache.Controller
+
+	// Store for the placements of the federated type
+	placementStore cache.Store
+	// Informer controller for placements of the federated type
+	placementController cache.Controller
```

Comment:
I didn't find the need for override and placement controllers. This rspcontroller is only creating/updating the override, placement objects as needed. Please remove these controllers.

Reply:
I do need it to trigger reconcile, in case the target override or placements are changed by a different entity (user, automated stuff) :).